lib/models/ObjectMDArchive.js

```diff
@@ -0,0 +1,172 @@
+/**
+ * class representing the archive of object metadata.
+ *
+ * @class
+ */
+class ObjectMDArchive {
+    /**
+     *
+     * @constructor
+     * @param {Object} archiveInfo contains the archive info set by the TLP and returned by the TLP jobs
+     * @param {Date} [restoreRequestedAt] set at the time restore request is made by the client
+     * @param {Number} [restoreRequestedDays] set at the time restore request is made by the client
+     * @param {Date} [restoreCompletedAt] set at the time of successful restore
+     * @param {Date} [restoreWillExpireAt] computed and stored at the time of restore
+     * @throws {Error} case of invalid parameter
+     */
+    constructor(
+        archiveInfo,
+        restoreRequestedAt = undefined,
+        restoreRequestedDays = undefined,
+        restoreCompletedAt = undefined,
+        restoreWillExpireAt = undefined,
```

Comment:
It's never really useful to explicitly write `undefined` because if you write it explicitly, it's no longer `undefined` as in "missing value". A general rule of thumb to make any inspection/debugging easier later for others is, if you want the value to be missing and the code *knows* that it's missing at that time use `null`, and if it's a value that the code was never aware of, it will automatically be `undefined` and that will be clear.

In this particular case, you can just ommit the `undefined` as it's the implicit value (which again does not make sense as explicit).

Reply:
Thanks, I udpated the function in https://github.com/scality/Arsenal/pull/1856/commits/11106e9db5c2a08f652d9ebf5564020125678863#diff-31d508ca346b0ada575d33da2a05b00e78a940bde0bc7b52e4c76408553fcc79R19
In the other PR for AmzRestore, https://github.com/scality/Arsenal/pull/1855/commits/da9090d3e20de3f102aa3a7f62274384ec80dc24#diff-f7e9b6b502cb91cc82d0dc4928a661ac320ea8b750c025c0cd759dd2142fca2fR19

I made sure not to edit the original commit from Fujifilm.